### PR TITLE
zh: docs/ops/configuration/traffic-management/dns/

### PR DESCRIPTION
**Pull request by rootsongjc**
Add the translation for [dns](http://localhost:1313/latest/docs/ops/configuration/traffic-management/dns/).

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

- [ ] Configuration Infrastructure
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by windsonsea** [comment]
Thanks. The lint error for your info:
```
./zh/docs/ops/configuration/traffic-management/dns/index.md:37: MD038 Spaces inside code span elements
./zh/docs/ops/configuration/traffic-management/dns/index.md:43: MD039 Spaces inside link text
```

File: content/zh/docs/ops/configuration/traffic-management/dns/index.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+本文档描述了底层实施细节。要了解更高层次的概述，请查看流量管理[概念](/zh/docs/concepts/traffic-management/)或[任务](/zh/docs/tasks/traffic-management/)页面。
+
+{{< /warning >}}
+
+## 请求过程{#life-of-a-request}
+
+下面我们将以一个应用程序运行 `curl example.com` 为例来看一个请求的全过程。这里的 `curl` 请求过程适用于几乎所有客户端。
+
+当你向一个域名发送请求时，客户端会进行 DNS 解析，将其解析为一个 IP 地址。不管 Istio 如何设置，这都会发生，因为 Istio只 是拦截网络流量；它不能改变应用程序的行为或决定是否发送 DNS 请求。在下面的例子中，`example.com` 被解析为 `192.0.2.0`。
```
Comment (windsonsea) [inline content/zh/docs/ops/configuration/traffic-management/dns/index.md:23]:
```suggestion
当您向一个域名发送请求时，客户端会进行 DNS 解析，将其解析为一个 IP 地址。不管 Istio 如何设置，这都会发生，因为 Istio 只是拦截网络流量；它不能改变应用程序的行为或决定是否发送 DNS 请求。在下面的例子中，`example.com` 被解析为 `192.0.2.0`。
```

File: content/zh/docs/ops/configuration/traffic-management/dns/index.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+{{< text bash >}}
+$ curl example.com -v
+*   Trying 192.0.2.0:80...
+{{< /text >}}
+
+接下来，该请求将被 Istio 拦截。这时，Istio 将看到主机名（来自 `Host: example.com` 头）和目标地址（`192.0.2.0:80`）。Istio 使用这些信息来确定预定目的地。理解流量路由对这种行为的工作原理进行了深入探讨。
```
Comment (windsonsea) [inline content/zh/docs/ops/configuration/traffic-management/dns/index.md:30]:
```suggestion
接下来，该请求将被 Istio 拦截。这时，Istio 将看到主机名（来自 `Host: example.com` 头）和目标地址（`192.0.2.0:80`）。Istio 使用这些信息来确定预定目的地。[理解流量路由](/docs/ops/configuration/traffic-management/traffic-routing/)对这种行为的工作原理进行了深入探讨。
```

File: content/zh/docs/ops/configuration/traffic-management/dns/index.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+如果客户端无法解析 DNS 请求，在 Istio 收到请求之前就会终止。这意味着，即使一个请求发送到一个 Istio 已知的主机名（例如，通过 `VirtualService` 配置），但是无法同过 DNS 解析，该请求也会失败。不过，Istio 的 [DNS 代理](#dns-proxing)可以改变这种行为。
```
Comment (windsonsea) [inline content/zh/docs/ops/configuration/traffic-management/dns/index.md:32]:
```suggestion
如果客户端无法解析 DNS 请求，在 Istio 收到请求之前就会终止。这意味着，即使一个请求发送到一个 Istio 已知的主机名（例如，通过 `VirtualService` 配置），但是无法通过 DNS 解析，该请求也会失败。不过 Istio 的 [DNS 代理](#dns-proxing)可以改变这种行为。
```

File: content/zh/docs/ops/configuration/traffic-management/dns/index.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+一旦 Istio 确定了预期的目的地，它必须选择要发送的地址。由于 Istio 的高级[负载均衡能力](/zh/docs/concepts/traffic-management/#load-balancing-options)，这往往不是客户端发送的原始 IP 地址。根据服务配置的不同，Istio 有几种不同的方式来实现：
```
Comment (windsonsea) [inline content/zh/docs/ops/configuration/traffic-management/dns/index.md:34]:
```suggestion
一旦 Istio 确定了预期的目的地，它必须选择要发送到的地址。由于 Istio 的高级[负载均衡能力](/zh/docs/concepts/traffic-management/#load-balancing-options)，这往往不是客户端发送的原始 IP 地址。根据服务配置的不同，Istio 有几种不同的方式来实现：
```

File: content/zh/docs/ops/configuration/traffic-management/dns/index.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+* 使用客户端的原始 IP 地址（`192.0.2.0`，在上面的例子中）。这种情况适用于 `resolution: NONE` 类型的 `ServiceEntry`  和[无头服务](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services)。
```
Comment (windsonsea) [inline content/zh/docs/ops/configuration/traffic-management/dns/index.md:36]:
```suggestion
* 使用客户端的原始 IP 地址（上例中为 `192.0.2.0`）。这种情况适用于 `resolution: NONE` 类型的 `ServiceEntry`  和[无头服务](https://kubernetes.io/zh-cn/docs/concepts/services-networking/service/#headless-services)。
```

File: content/zh/docs/ops/configuration/traffic-management/dns/index.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+* 在一组静态 IP 地址上进行负载均衡。这种情况适用于 `resolution: STATIC` 类型的 `ServiceEntry`，将使用其中所有的 `spec.endpoints` ，或者对于标准 `Service`，将使用所有 `Endpoint`。
```
Comment (windsonsea) [inline content/zh/docs/ops/configuration/traffic-management/dns/index.md:37]:
```suggestion
* 在一组静态 IP 地址上进行负载均衡。这种情况适用于 `resolution: STATIC` 类型的 `ServiceEntry`，将使用其中所有的 `spec.endpoints`，或者对于标准 `Service` 将使用所有 `Endpoint`。
```

File: content/zh/docs/ops/configuration/traffic-management/dns/index.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+* 使用 DNS 定期解析地址，并在所有结果中进行负载均衡。这种情况适用于 `resolution: DNS` 类型的 `ServiceEntry`。
+
+请注意，在任何情况下，Istio 代理内部的 DNS 解析与用户应用程序中的 DNS 解析是正交（orthogonal ）的。即使客户端进行了 DNS 解析，代理也可能忽略已解析的 IP 地址，而使用自己的地址，这些地址可能来自静态的 IP 列表或通过代理的 DNS 解析（可能是同一主机名或不同的主机名）。
+
+## 代理 DNS 解析{#proxying-dns-resolution}
+
+与大多数客户端在请求时按需执行 DNS 请求（然后通常缓存结果）不同，Istio 代理从不执行同步 DNS 请求。配置 `resolution: DNS` 类型的 `ServiceEntry` 后，代理将定期解析配置的主机名并将其用于所有请求。此时间间隔由DNS 响应的 [TTL](https://en.wikipedia.org/wiki/Time_to_live#DNS_records) 确定。即使代理从未向这些应用程序发送任何请求，该情况也会发生。
```
Comment (windsonsea) [inline content/zh/docs/ops/configuration/traffic-management/dns/index.md:44]:
```suggestion
与大多数客户端在请求时按需执行 DNS 请求（然后通常缓存结果）不同，Istio 代理从不执行同步 DNS 请求。配置 `resolution: DNS` 类型的 `ServiceEntry` 后，代理将定期解析配置的主机名并将其用于所有请求。此时间间隔由 DNS 响应的 [TTL](https://en.wikipedia.org/wiki/Time_to_live#DNS_records) 确定。即使代理从未向这些应用程序发送任何请求，该情况也会发生。
```

File: content/zh/docs/ops/configuration/traffic-management/dns/index.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+对于具有许多代理或许多 `resolution: DNS`  类型 `ServiceEntry` 的网格而言，尤其是在使用较低 `TTL` 时，可能会导致 DNS 服务器的负载很高。在这些情况下，以下行为有助于减轻负载：
```
Comment (windsonsea) [inline content/zh/docs/ops/configuration/traffic-management/dns/index.md:46]:
```suggestion
对于具有许多代理或许多 `resolution: DNS` 类型 `ServiceEntry` 的网格而言，尤其是在使用较低 `TTL` 时，可能会导致 DNS 服务器的负载很高。在这些情况下，以下行为有助于减轻负载：
```

**Review by windsonsea — COMMENTED** [review]
Overall look good to me. Pick some minor errors for improvement. Thanks.

File: content/zh/docs/ops/configuration/traffic-management/dns/index.md (hunk 2 above)
Comment (windsonsea) [reply content/zh/docs/ops/configuration/traffic-management/dns/index.md:30]: 嗯，链接需要先保留，后续对应的 zh 内容会陆续增加

Comment (rootsongjc) [reply content/zh/docs/ops/configuration/traffic-management/dns/index.md:30]: 这个链接指向了一个不存在的地址，如果加上去的话是否会导致这个 PR 校验失败？如果不会的话可以先加上。

Comment (windsonsea) [reply content/zh/docs/ops/configuration/traffic-management/dns/index.md:30]: 应该不会报错，试试呢

**Review by windsonsea — APPROVED** [review]

**Pull request merged**